### DISCUSSION TITLE
Changing ROOTFSPART_SIZE results in an error

### DISCUSSION THREAD

**Discussion by CHNtentes** [OE4T/meta-tegra#1189] [Q&A]
Hi. I'm using Orin NX with a custom board.

When I add ROOTFSPART_SIZE = "125627793408" (which is 117 GiB I believe) to conf file and build the image, it returns an error: 

> [  10.0109 ] End sector for APP, expected at: 124321758, actual: 0
[  10.0109 ] Error: Return value 4

The full log is here: [log.do_image_cpio.63284.txt](https://github.com/OE4T/meta-tegra/files/10839347/log.do_image_cpio.63284.txt)

What other files do I need to modify? I suppose there's a flash.xml or something like that, and 117 GiB exceeds the size of whole disk.

**Comment by madisongh** [comment]
Yeah, the NVIDIA tools check the partition sizing against the number of sectors on the device.  You should be able to solve this by creating a custom flash layout based off the default one, modifying the `num_sectors` property for the sdcard device to be the actual number of sectors on your NVMe drive. You might also want to change the `<allocation_attribute>` element for the APP partition to be `0x808` instead of just `0x8`, to have it automatically fill to the end of the drive (unless you want to reserve space for some other use).

**Reply by CHNtentes** [reply]
Thanks for your reply. May I ask where I can find the flash layout file? I searched under meta-tegra layer but can't find any.

**Reply by dwalkes** [reply]
@CHNtentes You can find an example of these in the mender layer, referenced from the instructions at https://github.com/OE4T/tegra-demo-distro/wiki/Customization-of-Storage-Size-and-root-filesystem-size-on-Mender-Builds.  The flash layout files come from NVIDIA and are modified for use with the mender layer.  You can do a similar thing with your layer

**Reply by dwalkes** [reply]
You ultimately need to override the `PARTITION_FILE` variable as done here: https://github.com/mendersoftware/meta-mender-community/blob/40bbc57035cbb5c8910cd798c4efa4b93c43ff17/meta-mender-tegra/recipes-bsp/tegra-binaries/tegra-bootfiles_%25.bbappend#L3

**Reply by CHNtentes** [reply]
@dwalkes Do I need to modify the `PARTITION_FILE_EXTERNAL` variable too? Since I'm using Orin NX with nvme storage.

**Reply by dwalkes** [reply]
If I understand the logic at https://github.com/OE4T/meta-tegra/commit/f38be1da19ee45acad468826a9deb0f8543e0310 and https://github.com/OE4T/meta-tegra/blob/88f299a9f73eda3dd5b66ac6fb61fb953bf40ac9/recipes-bsp/tegra-binaries/tegra-bootfiles_35.2.1.bb it looks like it will try to use `PARTITION_FILE` as the source for partition sizes on Orin NX with nvme.  I haven't experimented with this myself, however.
